### DISCUSSION TITLE
How to force a timeout?

### DISCUSSION THREAD

**Discussion by vitidev** [encode/httpx#1778] [Q&A]
The documentation says that TimeoutException raises after a given period of inactivity.
But what to do if there is no inactivity? - the file downloads slowly, but downloads and no TimeoutException occurs 

for example

       timeout = httpx.Timeout(1.0)
       r = await client.get(url, timeout=timeout)

will work until everything downloads 
In this case aiohttp raise TimeoutError after 1 seconds

**Comment by lovelydinosaur** [comment]
We don't currently have any timeout parameter for the entire response.

There's some discussion on adding further resource limits, here... https://github.com/encode/httpx/issues/1450#issuecomment-758195782

**Comment by lewoudar** [comment]
@vitidev I think you can do something like this. I generally prefer this pattern instead of configuring x timeouts.

```python
import anyio
import httpx


async def main():
    with anyio.move_on_after(1.0) as scope:
        async with httpx.AsyncClient(base_url='https://example.com', timeout=None) as client:
            print(await client.get('/'))

    print('scope cancelled:', scope.cancel_called)


anyio.run(main)
```
